src/components/widgets/support-case-table-filter.tsx

```diff
@@ -0,0 +1,312 @@
+        severity: filters.severity,
+        status: filters.status.filter((fil: string) => fil !== id),
+      });
+    } else {
+      setFilters({ productFamily: [], severity: [], status: [] });
+    }
+  };
+
+  const onDeleteGroup = (type: string) => {
+    if (type === 'Product Family') {
+      setFilters({
+        productFamily: [],
+        severity: filters.severity,
+        status: filters.status,
+      });
+    } else if (type === 'Severity') {
+      setFilters({
+        productFamily: filters.productFamily,
+        severity: [],
+        status: filters.status,
+      });
+    } else if (type === 'Status') {
+      setFilters({
+        productFamily: filters.productFamily,
+        severity: filters.severity,
+        status: [],
+      });
+    }
+  };
+
+  const onProductFamilyToggle = () => {
+    setIsProductFamilyExpanded(!isProductFamilyExpanded);
+  };
+
+  const onSeverityToggle = () => {
+    setIsSeverityExpanded(!isSeverityExpanded);
+  };
+
+  const onStatusToggle = () => {
+    setIsStatusExpanded(!isStatusExpanded);
+  };
+
+  const statusMenuItems = (
+    <SelectList>
+      <SelectOption
+        hasCheckbox
+        key={statusTypes.closed}
+        value={statusTypes.closed}
+        isSelected={filters.status.includes(statusTypes.closed)}
+      >
+        {statusTypes.closed}
+      </SelectOption>
+      <SelectOption
+        hasCheckbox
+        key={statusTypes.customerWaiting}
+        value={statusTypes.customerWaiting}
+        isSelected={filters.status.includes(statusTypes.customerWaiting)}
+      >
+        {statusTypes.customerWaiting}
+      </SelectOption>
+      <SelectOption
+        hasCheckbox
+        key={statusTypes.redHatWaiting}
+        value={statusTypes.redHatWaiting}
+        isSelected={filters.status.includes(statusTypes.redHatWaiting)}
+      >
+        {statusTypes.redHatWaiting}
+      </SelectOption>
+    </SelectList>
+  );
+
+  const severityMenuItems = (
+    <SelectList>
+      <SelectOption
+        hasCheckbox
+        key={severityTypes.low}
+        value={severityTypes.low}
+        isSelected={filters.severity.includes(severityTypes.low)}
+      >
+        {severityTypes.low}
+      </SelectOption>
+      <SelectOption
+        hasCheckbox
+        key={severityTypes.normal}
+        value={severityTypes.normal}
+        isSelected={filters.severity.includes(severityTypes.normal)}
+      >
+        {severityTypes.normal}
+      </SelectOption>
+      <SelectOption
+        hasCheckbox
+        key={severityTypes.high}
+        value={severityTypes.high}
+        isSelected={filters.severity.includes(severityTypes.high)}
+      >
+        {severityTypes.high}
+      </SelectOption>
+      <SelectOption
+        hasCheckbox
+        key={severityTypes.urgent}
+        value={severityTypes.urgent}
+        isSelected={filters.severity.includes(severityTypes.urgent)}
+      >
+        {severityTypes.urgent}
+      </SelectOption>
+    </SelectList>
+  );
```

Comment:
This could be definitely deduplicated - generated just from array of severity variants